### ISSUE TITLE
Curious on why  `simple_xml::Error` doesn't implement `Error`.

### ISSUE THREAD

**Issue by OrangeTux** [ten3roberts/simple-xml#3]
Hi, 

thanks for this library. I noticed that [`simple_xml::Error`](https://github.com/ten3roberts/simple-xml/blob/master/src/error.rs#L2-L8) doesn't implemented the [`Error`](https://doc.rust-lang.org/std/error/trait.Error.html) trait. That makes incompatible with crates like [anyhow](https://docs.rs/anyhow/latest/anyhow/index.html). Is a deliberate decision not to implement `Error` for this type? If so, can you elaborate on this choice?

**Comment by ten3roberts** [comment]
No, there's no reason for it. I was quite new to Rust when writing this (one of my first ever crates).

I've added it now, let me know if it is satisfactory :)

**Comment by OrangeTux** [comment]
Thanks for implementing the trait :)

The implementation looks solid. You could consider implementing the trait yourself instead of relying of `thiserror` to reduce the number of dependencies.
